### PR TITLE
Add userSettings section for helping deployer use config API

### PR DESCRIPTION
**Pull request by paulczar**
This makes it a little bit easier for the person configuring Harbor as it will provide a place to put user settings to be displayed in NOTES.txt for hints on using the config api via curl.

Signed-off-by: Paul Czarkowski <username.taken@gmail.com>

### CODE REVIEW

**Comment by tcarmet** [comment]
Hello @paulczar and @ywk253100 

I already have a working patch that creates a Kubernetes `Job` with the same command.

I can happily contribute it on top of your changes if you would like to go forward on that direction.

Just let me know :)

**Comment by ywk253100** [comment]
I like this idea, thanks @tcarmet.
@paulczar I think this is a better option, what do you think?

**Comment by paulczar** [comment]
I think running `curl` commands at the API could be a bit fragile to automate.  I'd rather see work on a harbor-cli that could be run to set those configs, I've seen a proposal for a harbor-cli in the community repo.

I'd rather just be able to set this stuff via a config file or environment variable as that's the way most software is configured, and most automation tooling handles this sort of thing.  Introducing this configuration API without a decent client to use is making it much harder to automate.

**Comment by Hokwang** [comment]
My opinion is that both of yours are not easy.
I just want to 
simply user can add/edit some vaules in values.yaml
and it works directly, not curl or not harbor-cli, etc.

like codes in core-cm.yaml
https://github.com/goharbor/harbor-helm/blob/master/templates/core/core-cm.yaml#L47-L51

**Comment by paulczar** [comment]
@Hokwang I agree!  See this issue - https://github.com/goharbor/harbor/issues/8076

File: templates/NOTES.txt [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
-For more details, please visit https://github.com/goharbor/harbor.
\ No newline at end of file
+For more details, please visit https://github.com/goharbor/harbor.
+
+{{ if .Values.userSettings }}
+If this is your first time Installing Harbor,
+You may need to use the Harbor Configuration API to
+finish your configuration by running:
+
+$ curl -i -X PUT -u "admin:{{ .Values.harborAdminPassword }}" \
```
Comment (hobti01) [inline templates/NOTES.txt]: I like the idea of the hint, but revealing the admin password in output that could be going to a pipeline deployment is not a secure idea.

File: templates/NOTES.txt [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
+  -H "Content-Type: application/json" \
+  {{ .Values.externalURL }}/api/configurations \
+  -d '{{ .Values.userSettings | toJson }}'
```
Comment (hobti01) [inline templates/NOTES.txt]: Same concern here, dumping the client secrets and other sensitive data to deployment output is a risk.

**Pull request closed without merging**